From 2484c00b8394a60411ea585b8b4ddf76a9ab77c9 Mon Sep 17 00:00:00 2001
From: Paolo Valente <paolo.valente@linaro.org>
Date: Mon, 3 Feb 2020 10:19:50 +0100
Subject: [PATCH 10/15] block, bfq: remove ifdefs from around gets/puts of bfq
 groups

ifdefs around gets and puts of bfq groups reduce readability, remove them.

Reported-by: Jens Axboe <axboe@kernel.dk>
Signed-off-by: Paolo Valente <paolo.valente@linaro.org>
---
 block/bfq-cgroup.c  | 4 ++++
 block/bfq-iosched.c | 6 ++----
 block/bfq-iosched.h | 1 +
 3 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/block/bfq-cgroup.c b/block/bfq-cgroup.c
index a29cf5c0b..f9c721f1c 100644
--- a/block/bfq-cgroup.c
+++ b/block/bfq-cgroup.c
@@ -1465,6 +1465,10 @@ struct bfq_group *bfqq_group(struct bfq_queue *bfqq)
 	return bfqq->bfqd->root_group;
 }
 
+void bfqg_and_blkg_get(struct bfq_group *bfqg) {}
+
+void bfqg_and_blkg_put(struct bfq_group *bfqg) {}
+
 struct bfq_group *bfq_create_group_hierarchy(struct bfq_data *bfqd, int node)
 {
 	struct bfq_group *bfqg;
diff --git a/block/bfq-iosched.c b/block/bfq-iosched.c
index 8aa92a9f4..637b22381 100644
--- a/block/bfq-iosched.c
+++ b/block/bfq-iosched.c
@@ -5405,9 +5405,7 @@ void bfq_put_queue(struct bfq_queue *bfqq)
 {
 	struct bfq_queue *item;
 	struct hlist_node *n;
-#ifdef CONFIG_BFQ_GROUP_IOSCHED
 	struct bfq_group *bfqg = bfqq_group(bfqq);
-#endif
 
 	assert_spin_locked(&bfqq->bfqd->lock);
 
@@ -5491,8 +5489,8 @@ void bfq_put_queue(struct bfq_queue *bfqq)
 
 #ifdef CONFIG_BFQ_GROUP_IOSCHED
 	bfq_log_bfqq(bfqq->bfqd, bfqq, "putting blkg and bfqg %p\n", bfqg);
-	bfqg_and_blkg_put(bfqg);
 #endif
+	bfqg_and_blkg_put(bfqg);
 	if (bfqq->bfqd)
 		bfq_log_bfqq(bfqq->bfqd, bfqq, "%p freed", bfqq);
 
@@ -7134,10 +7132,10 @@ static void bfq_exit_queue(struct elevator_queue *e)
 
 	BFQ_BUG_ON(hrtimer_active(&bfqd->idle_slice_timer));
 
-#ifdef CONFIG_BFQ_GROUP_IOSCHED
 	/* release oom-queue reference to root group */
 	bfqg_and_blkg_put(bfqd->root_group);
 
+#ifdef CONFIG_BFQ_GROUP_IOSCHED
 	blkcg_deactivate_policy(bfqd->queue, &blkcg_policy_bfq);
 #else
 	spin_lock_irq(&bfqd->lock);
diff --git a/block/bfq-iosched.h b/block/bfq-iosched.h
index 965e7432d..696b5ca41 100644
--- a/block/bfq-iosched.h
+++ b/block/bfq-iosched.h
@@ -927,6 +927,7 @@ struct bfq_group {
 
 #else
 struct bfq_group {
+	struct bfq_entity entity;
 	struct bfq_sched_data sched_data;
 
 	struct bfq_queue *async_bfqq[2][IOPRIO_BE_NR];
-- 
2.25.0.191.gde93cc14ab.dirty

